### PR TITLE
Feature/improved esim batch management

### PR DESCRIPTION
**Pull request by la3lma**
This is the first usable version of the sim batch-management command line tool.   

It is  a _huuuge_ commit, so I don't expect any reviewer to look at all of it. What I would like, very much, is to get a little "shotgun review" where you just pick a part, perhaps even a very small part,  and comment on what you like or dislike.  I will then extrapolate your comments to other locations where they may be applicable.

In short: I'm looking for quality, rather than quantity in the review :-)

@gordonklaus  the previous PR you reviewed (the es2+ stack) is deleted, but only after the revised version was merged into this branch.

### CODE REVIEW

File: sim-administration/sim-batch-management/uploadtoprime/uploadtoprime.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package uploadtoprime
+
+import (
+	"fmt"
+	"github.com/ostelco/ostelco-core/sim-administration/sim-batch-management/model"
+	"github.com/ostelco/ostelco-core/sim-administration/sim-batch-management/store"
+	"strings"
+)
+
+func GeneratePostingCurlscript(url string, payload string) {
```
Comment (la3lma) [inline sim-administration/sim-batch-management/uploadtoprime/uploadtoprime.go:13]: Add comments describing in detail what these things are doing, and why.

File: sim-administration/sim-batch-management/store/store_test.go [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+func TestMain(m *testing.M) {
+	setup()
+	code := m.Run()
+	shutdown()
+	os.Exit(code)
+}
+
+func setup() {
+
+	// In memory database fails, so we try this gonzo method of getting
```
Comment (la3lma) [inline sim-administration/sim-batch-management/store/store_test.go:27]: Typo

File: sim-administration/sim-batch-management/store/store_test.go [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+	// a fresh database
+	filename := "bazunka.db"
```
Comment (la3lma) [inline sim-administration/sim-batch-management/store/store_test.go:29]: Use a less creative name, e.g. ."test-sqlite-database-for-storage-testing.db"

**Review by la3lma — COMMENTED** [review]
Self-review :). Some things caught, I'm sure there is more there.
